packages/mui-styles/test/theme-scoping.test.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from 'react';
+import { expect } from 'chai';
+import { createRenderer } from 'test/utils';
+import * as material from '@mui/material/styles';
+import * as joy from '@mui/joy/styles';
+import { makeStyles } from '@mui/styles';
+
+describe('Theme scoping', () => {
+  const { render } = createRenderer();
+
+  it('works without theme scoping', () => {
+    const useStyles = makeStyles<material.Theme>((theme) => ({
+      root: {
+        color: theme.palette.primary.main,
+      },
+    }));
+    function Component() {
+      const classes = useStyles();
+
+      return <div className={classes.root}>Component</div>;
+    }
+
+    expect(() =>
+      render(
+        <material.ThemeProvider theme={material.createTheme()}>
+          <Component />
+        </material.ThemeProvider>,
+      ),
+    ).not.to.throw();
+  });
+
+  it('theme scoping works', () => {
+    const useStyles = makeStyles<material.Theme>((theme) => ({
+      root: {
+        color: theme.palette.primary.main,
+      },
+    }));
+    function Component() {
+      const classes = useStyles();
+
+      return <div className={classes.root}>Component</div>;
+    }
+
+    expect(() =>
+      render(
+        <material.ThemeProvider theme={{ [material.THEME_ID]: material.createTheme() }}>
+          <Component />
+        </material.ThemeProvider>,
+      ),
+    ).not.to.throw();
```

Comment:
We could ensure that the correct color is used too, for e.g. using a custom theme.

Reply:
Added.